src/components/Reports/StaffExpenseReport/DownloadButtonGroup/downloadReport.tsx

```diff
@@ -0,0 +1,87 @@
+import { TFunction } from 'i18next';
+import { buildURI } from 'react-csv/lib/core';
+import { Transaction } from 'src/components/Reports/StaffExpenseReport/StaffExpenseReport';
+import { ReportType } from '../Helpers/StaffReportEnum';
+
+const makeTable = (
+  title: string,
+  csvHeader: string[],
+  transactions: Transaction[],
+  locale: string,
+) => {
+  const csvData = [
+    [title],
+    csvHeader,
+    ...transactions.map((transaction) => [
+      transaction.month,
+      transaction.category,
+      transaction.total.toLocaleString(locale, {
+        style: 'currency',
+        currency: 'USD',
+      }),
+    ]),
+  ];
+
+  return csvData;
+};
+
+function createCombinedReport(
+  transactions: Transaction[],
+  titles: { income: string; expense: string; combined: string },
+  csvHeader: string[],
+  locale: string,
+) {
+  const income = transactions.filter((transaction) => transaction.total > 0);
+  const expenses = transactions.filter((transaction) => transaction.total < 0);
+  const incomeData = makeTable(
+    titles[ReportType.Income],
+    csvHeader,
+    income,
+    locale,
+  );
+  const expenseData = makeTable(
+    titles[ReportType.Expense],
+    csvHeader,
+    expenses,
+    locale,
+  );
+  return [...incomeData, [''], ...expenseData];
+}
+
+export const downloadCsv = (
+  type: ReportType,
+  enqueueSnackbar: (message: string, options?: { variant: string }) => void,
+  transactions: Transaction[],
+  t: TFunction,
+  locale: string,
+) => {
+  if (!transactions || transactions.length === 0) {
+    enqueueSnackbar(t('No transactions to download'), { variant: 'error' });
+    return;
+  }
+
+  const titles = {
```

Comment:
I think it would be better to not use an object lookup here, and inline the `t` call, i.e. `t('Income Report')`.